### PR TITLE
feat: Add support for AWS Secrets Manager

### LINKED ISSUE #2216: Support for AWS Secrets Manager
## Is your feature request related to a problem? Please describe.

I have personal secrets in 1password, but my company has secrets in AWS Secrets Manager. I'd like to have my dotfiles pull from both places natively. I know I could use the secret command line callout, but having something natively integrated is nice.

## Describe the solution you'd like

I would like some template functions that allow for reading secrets from AWS Secrets Manager.

## Describe alternatives you've considered

Could use the secret command callout, which is likely sufficient, but it's not native.

## Additional context

If this is of interest, I have it working locally, just need to add documentation and figure out how to do testing, as not everyone who is running tests will have secrets in AWS...

It does take a dependency on the `aws-go-sdk-v2`, which might be a deal breaker if `chezmoi` doesn't want further dependencies. It is natively go, so it's just like any other dep.

### PR DESCRIPTION
**Pull request by craiggwilson**
Fixes #2216 

This uses the `aws-go-sdk-v2` to add native support for [AWS Secrets Manager](https://aws.amazon.com/secrets-manager/). On my machine (WSL ubuntu), this takes the binary size from ~28MB to ~31MB.

### CODE REVIEW

File: assets/chezmoi.io/docs/reference/configuration-file/variables.md [hunk 1]
```diff
@@ -2,78 +2,81 @@
...
+|                     | `recipient`           | string   | *none*                   | age recipient                                          |
+|                     | `recipients`          | []string | *none*                   | age recipients                                         |
+|                     | `recipientsFile`      | []string | *none*                   | age recipients file                                    |
+|                     | `recipientsFiles`     | []string | *none*                   | age recipients files                                   |
+|                     | `suffix`              | string   | `.age`                   | Suffix appended to age-encrypted files                 |
+|                     | `symmetric`           | bool     | `false`                  | Use age symmetric encryption                           |
+| `awsSecretsManager` | `profile`             | string   | *none*                   | The AWS shared profile name                            |
+|                     | `region`              | string   | *none*                   | The AWS region                                         |
```
Comment (craiggwilson) [inline assets/chezmoi.io/docs/reference/configuration-file/variables.md:34]: These were the only changes here.

File: pkg/cmd/config.go [hunk 2]
```diff
@@ -91,15 +91,16 @@ type ConfigFile struct {
...
-	Bitwarden   bitwardenConfig   `mapstructure:"bitwarden"`
-	Gopass      gopassConfig      `mapstructure:"gopass"`
-	Keepassxc   keepassxcConfig   `mapstructure:"keepassxc"`
-	Keeper      keeperConfig      `mapstructure:"keeper"`
-	Lastpass    lastpassConfig    `mapstructure:"lastpass"`
-	Onepassword onepasswordConfig `mapstructure:"onepassword"`
-	Pass        passConfig        `mapstructure:"pass"`
-	Secret      secretConfig      `mapstructure:"secret"`
-	Vault       vaultConfig       `mapstructure:"vault"`
+	AwsSecretsManager awsSecretsManagerConfig `mapstructure:"awsSecretsManager"`
```
Comment (craiggwilson) [inline pkg/cmd/config.go:94]: This is the only change here.

Comment (twpayne) [reply pkg/cmd/config.go:94]: Tiny nit: please rename this field to `AWSSecretManager` (https://github.com/golang/go/wiki/CodeReviewComments#initialisms).

**Review by twpayne — COMMENTED** [review]
This is as near a perfect PR as I could possibly imagine! Thank you!

As there's a tiny misspelling, could you also change the field name as commented?

But seriously, this is fantastic, thank you!

**Comment by twpayne** [comment]
Absolutely fantastic, thank you :)

**Pull request merged**